operator/pkg/operator.go

```diff
@@ -136,6 +149,51 @@ func (o *Operator) SubscribeToNewTasksV3() (chan error, error) {
 	return o.avsSubscriber.SubscribeToNewTasksV3(o.NewTaskCreatedChanV3)
 }
 
+type OperatorLastProcessedBatch struct {
+	BlockNumber uint32 `json:"block_number"`
+}
+
+func (o *Operator) LoadLastProcessedBatch() error {
+	file, err := os.ReadFile(o.lastProcessedBatchLogFile)
+
+	if err != nil {
+		return fmt.Errorf("failed read from file: %v", err)
+	}
+
+	err = json.Unmarshal(file, &o.lastProcessedBatch)
+
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal batch: %v", err)
+	}
+
+	return nil
+}
+
+func (o *Operator) UpdateLastProcessBatch(blockNumber uint32) error {
+	// we want to store the latest block number
+	if blockNumber < o.lastProcessedBatch.BlockNumber {
+		return nil
+	}
+
+	o.lastProcessedBatch = OperatorLastProcessedBatch{BlockNumber: blockNumber}
+
+	// write to a file so it can be recovered in case of operator outage
+	json, err := json.Marshal(o.lastProcessedBatch)
+
+	if err != nil {
+		return fmt.Errorf("failed to marshal batch: %v", err)
+	}
+
+	err = os.WriteFile(o.lastProcessedBatchLogFile, json, os.ModePerm)
+	if err != nil {
+		return fmt.Errorf("failed to write to file: %v", err)
+	}
+
+	o.Logger.Info("Updated latest block json file")
```

Comment:
Consider also printing the block number here.

Reply:
Updated!